LLMonFHIR/FHIR Views/FHIRResourcesView.swift

```diff
@@ -124,24 +125,93 @@ struct FHIRResourcesView<ContentView: View, ActionView: View>: View {
     
     
     private func section(for keyPath: KeyPath<FHIRStore, [FHIRResource]>, sectionName: String) -> some View {
-        var resources = fhirStore[keyPath: keyPath]
-        
-        if !searchText.isEmpty {
-            resources = resources.filterByDisplayName(with: searchText)
-        }
-        
+        let resources = filteredResources(for: keyPath)
+
         guard !resources.isEmpty else {
             return AnyView(EmptyView())
         }
-        
+
+        let showAll = Binding(
+            get: { showAllItems[sectionName, default: false] },
+            set: { showAllItems[sectionName] = $0 }
+        )
+
         return AnyView(
-            Section(sectionName) {
-                ForEach(resources.sorted(by: { ($0.date ?? .distantPast) > ($1.date ?? .distantPast) })) { resource in
-                    NavigationLink(value: resource) {
-                        FHIRResourceSummaryView(resource: resource)
-                    }
-                }
+            Section {
+                sectionHeaderButton(sectionName: sectionName, resources: resources, showAll: showAll)
+                resourcesList(resources: resources, showAll: showAll)
+            } header: {
+                EmptyView()
```

Comment:
Wondering if we can actually use the header here to make it a bit more similar to default system sections that are expandable (e.g. in mail).

Reply:
Like this?

<img width="400" alt="Simulator Screenshot - iPhone 16 Plus - 2025-03-12 at 23 48 56" src="https://github.com/user-attachments/assets/644a3a9c-f086-45e1-b16d-d693a78c7414" />

Reply:
That looks nice! 🚀